client/authority-discovery/src/worker.rs

```diff
@@ -328,18 +328,30 @@ where
 			Role::Discover => HashSet::new(),
 		};
 
-		let mut authorities = self
+		let authorities = self
 			.client
 			.runtime_api()
 			.authorities(&id)
 			.map_err(Error::CallingRuntime)?
 			.into_iter()
-			.filter(|id| !local_keys.contains(id.as_ref()))
-			.collect();
+			.filter(|id| !local_keys.contains(id.as_ref()));
+		let next_authorities = self
+			.client
+			.runtime_api()
+			.next_authorities(&id)
+			.unwrap_or_default()
```

Comment:
the runtime error here is ignored, I don't know whether we will need that when introduced on a live chain or whether we will need a migration